File: android/src/main/java/com/pinmi/react/printer/adapter/abstracts/PrinterAdapter.java
package com.pinmi.react.printer.adapter.abstracts;

import android.graphics.Bitmap;

import com.facebook.react.bridge.Callback;
import com.facebook.react.bridge.ReactApplicationContext;

import java.util.List;

/**
 * Created by xiesubin on 2017/9/21.
 */

public interface PrinterAdapter {


    public void init(ReactApplicationContext reactContext, Callback successCallback, Callback errorCallback);

    public List<PrinterDevice> getDeviceList(Callback errorCallback);

    public void selectDevice(PrinterDeviceId printerDeviceId, Callback successCallback, Callback errorCallback);

    public void closeConnectionIfExists();

    public void printRawData(String rawBase64Data, Callback errorCallback);

    public void printImageData(String imageUrl, int imageWidth, int imageHeight, Callback errorCallback);

    public void printImageBase64(Bitmap imageUrl, int imageWidth, int imageHeight, Callback errorCallback);
    public void getStatus(Callback callback);
}

File: android/src/main/java/com/pinmi/react/printer/adapter/abstracts/PrinterAdapter.kt
package com.pinmi.react.printer.adapter.abstracts

import android.graphics.Bitmap
import com.facebook.react.bridge.Callback
import com.facebook.react.bridge.ReactApplicationContext

/**
 * Created by misaeldossantos on 2023/02/25.
 */
interface PrinterAdapter {
    fun init(reactContext: ReactApplicationContext?, successCallback: Callback, errorCallback: Callback?)
    fun getDeviceList(errorCallback: Callback): List<PrinterDevice>
    fun selectDevice(printerDeviceId: PrinterDeviceId, successCallback: Callback, errorCallback: Callback)
    fun closeConnectionIfExists()
    fun printRawData(rawBase64Data: String, errorCallback: Callback)
    fun printImageData(imageUrl: String?, imageWidth: Int, imageHeight: Int)
    fun printImageBase64(imageUrl: Bitmap?, imageWidth: Int, imageHeight: Int)
    fun getStatus(): String
}
